src/JMS/Serializer/GraphNavigator.php

```diff
@@ -292,14 +292,18 @@ private function resolveMetadata($data, ClassMetadata $metadata)
                 break;
 
             default:
-                throw new \LogicException(sprintf(
-                    'The discriminator field name "%s" for base-class "%s" was not found in input data.',
-                    $metadata->discriminatorFieldName,
-                    $metadata->name
-                ));
+                if($metadata->discriminatorStrictDeserialize) {
+                    throw new \LogicException( sprintf(
+                        'The discriminator field name "%s" for base-class "%s" was not found in input data.',
+                        $metadata->discriminatorFieldName,
+                        $metadata->name
+                    ) );
+                } else {
+                    $typeValue = null;
```

Comment:
instead of this you can just `return $metadata;` and you will not need the changes on line 306